src/components/app/app.test.js

```diff
@@ -67,4 +68,22 @@ describe('App', () => {
       expect(() => shallow(<App />)).toThrow();
     });
   });
+
+  it("resets the active pipeline when data prop is updated, if the active pipeline is not included in the new dataset's list of pipelines", () => {
+    // Find a pipeline that is in the first dataset but not the second
+    const activePipeline = animals.pipelines.find(
+      pipeline => !demo.pipelines.map(d => d.id).includes(pipeline.id)
+    );
+    const { container, rerender } = render(<App data={animals} />);
+    const pipelineDropdown = container.querySelector('.pipeline-list');
+    const menuOption = within(pipelineDropdown).getByText(activePipeline.name);
+    const pipelineDropdownLabel = pipelineDropdown.querySelector(
+      '.kui-dropdown__label > span:first-child'
+    );
+    expect(pipelineDropdownLabel.innerHTML).toBe('Default');
```

Comment:
Totally agree with you: God I tried to use something else, I really did. It's something I wanted to flag: This dropdown has no label, and few roles, etc. I'm trying to select the element and enquire about what its text is, so I can't select it by its text. I tried using `innerText` but that returned `undefined` for some reason, but `innerHTML` worked so I went with it.

I just worked out that I could do this instead:
```es6
    const pipelineDropdownLabel = within(pipelineDropdown)
      .getByRole('button')
      .querySelector('span:first-child');
    expect(getNodeText(pipelineDropdownLabel)).toBe('Default');
    fireEvent.click(menuOption);
    expect(getNodeText(pipelineDropdownLabel)).toBe(activePipeline.name);
    rerender(<App data={demo} />);
    expect(getNodeText(pipelineDropdownLabel)).toBe('Default');
  });
```
but it's pretty much the same anyway? Happy to change it though.

Reply:
Fair enough, text nodes can make things tricky. It looks like https://github.com/testing-library/jest-dom is the recommended way in this case, which might be worth trying in future if we don't have it ready, but for now this is fine since it's in a test script only.

As for the label, I'd assume [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute) will be the cleanest way to add that?